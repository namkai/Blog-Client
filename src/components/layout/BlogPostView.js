import React, { PureComponent } from 'react';

class BlogPostView extends PureComponent {
	render() {
		return (
			<div className="container pt-4">
				<div className="row">
					<div className="col-lg-3">
						{ this.props.card }
					</div>
					<div className="col-lg-6">
						<ul className="list-group media-list media-list-stream mb-4">
							{ this.props.children }
						</ul>
					</div>
				</div>
			</div>
		);
	}
}

export default BlogPostView;