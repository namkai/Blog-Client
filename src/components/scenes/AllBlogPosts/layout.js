import React, { PureComponent } from 'react';

class BlogPostLayout extends PureComponent {
	render() {
		return (
			<div className="container pt-4">
				<div className="row">
					<div className="col-lg-2">
						{/*{ this.props.card }*/}
					</div>
					<div className="col-lg-8">
						<ul className="list-group media-list media-list-stream mb-4">
							{ this.props.children }
						</ul>
					</div>
				</div>
			</div>
		);
	}
}

export default BlogPostLayout;