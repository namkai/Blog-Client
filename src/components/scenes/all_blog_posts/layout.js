import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';

export class BlogPostLayout extends PureComponent {
	render() {
		return (
			<div className="container pt-4">
				<div className="row">
					<div className="col-lg-2">
						{/*{ this.props.card }*/}
					</div>
					<div className="col-lg-8">
						<ul className="list-group media-list media-list-stream mb-4">
							{ this.props.children }
						</ul>
					</div>
				</div>
			</div>
		);
	}
}

BlogPostLayout.defaultProps = {
	children: [],
};

BlogPostLayout.propTypes = {
	children: PropTypes.array,
};

export default BlogPostLayout;