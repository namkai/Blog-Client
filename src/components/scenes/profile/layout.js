import PropTypes from 'prop-types';
import React from 'react';
import { Link } from 'react-router-dom';
import '../../../style/profile.css';

export const ProfileLayout = ({ name, backgroundPhoto, profilePhoto, children }) => (
	<div className="profile-header text-center" style={{ backgroundImage: `url(${backgroundPhoto})` }}>
		<div className="container-fluid">
			<div className="container-inner">
				<img className="rounded-circle media-object" src={profilePhoto}/>
				<h3 className="profile-header-user">{name}</h3>
				<p className="profile-header-bio">Check out my blog posts</p>
			</div>
		</div>
		<nav className="profile-header-nav">
			<ul className="nav nav-tabs justify-content-center">
				<li className="nav-item active">
					<Link className="nav-link" to="/profile/posts">Blog Posts</Link>
				</li>
			</ul>
		</nav>
		{children}
	</div>
);

ProfileLayout.defaultProps = {
	name: '',
	backgroundPhoto: '',
	profilePhoto: '',
	children: [],
};

ProfileLayout.defaultProps = {
	name: PropTypes.string,
	backgroundPhoto: PropTypes.string,
	profilePhoto: PropTypes.string,
	children: PropTypes.array,
};

export default ProfileLayout;
