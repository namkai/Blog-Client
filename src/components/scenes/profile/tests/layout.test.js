import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import React from 'react';
import { ProfileLayout } from '../layout';

it('renders correctly', () => {
	const layout = shallow(
		<ProfileLayout />,
	);
	expect(toJson(layout)).toMatchSnapshot();
});
