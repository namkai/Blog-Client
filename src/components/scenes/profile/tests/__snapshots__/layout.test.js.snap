// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="profile-header text-center"
  style={
    Object {
      "backgroundImage": "url(function () { [native code] })",
    }
  }
>
  <div
    className="container-fluid"
  >
    <div
      className="container-inner"
    >
      <img
        className="rounded-circle media-object"
        src={[Function]}
      />
      <h3
        className="profile-header-user"
      />
      <p
        className="profile-header-bio"
      >
        Check out my blog posts
      </p>
    </div>
  </div>
  <nav
    className="profile-header-nav"
  >
    <ul
      className="nav nav-tabs justify-content-center"
    >
      <li
        className="nav-item active"
      >
        <Link
          className="nav-link"
          replace={false}
          to="/profile/posts"
        >
          Blog Posts
        </Link>
      </li>
    </ul>
  </nav>
</div>
`;
